tests/test_main.py

```diff
@@ -132,3 +133,138 @@ async def test_run(queries, expected_response, store_errors, exception):
+    with NamedTemporaryFile("w") as output_file:
+        with pytest.warns(
+            DeprecationWarning, match=r"^The stop_on_errors parameter is deprecated\.$"
+        ):
+            with pytest.raises(RequestError):
+                await run(
+                    queries=queries,
+                    out=output_file,
+                    n_conn=1,
+                    api_url=mockserver.urljoin("/"),
+                    api_key="a",
+                    stop_on_errors=True,
+                )
+
+
+def _run(*, input, mockserver, cli_params=None):
+    cli_params = cli_params or tuple()
+    with NamedTemporaryFile("w") as url_list:
+        url_list.write(input)
+        url_list.flush()
+        # Note: Using “python -m zyte_api” instead of “zyte-api” enables
+        # coverage tracking to work.
+        result = subprocess.run(
+            [
+                "python",
+                "-m",
+                "zyte_api",
+                "--api-key",
+                "a",
+                "--api-url",
+                mockserver.urljoin("/"),
+                url_list.name,
+                *cli_params,
+            ],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+    return result
+
+
+def test_empty_input(mockserver):
+    result = _run(input="", mockserver=mockserver)
+    assert result.returncode
+    assert result.stdout == b""
+    assert result.stderr == b"No input queries found. Is the input file empty?\n"
+
+
+def test_intype_txt_implicit(mockserver):
+    result = _run(input="https://a.example", mockserver=mockserver)
+    assert not result.returncode
+    assert (
+        result.stdout
+        == b'{"url": "https://a.example", "browserHtml": "<html><body>Hello<h1>World!</h1></body></html>"}\n'
+    )
+
+
+def test_intype_txt_explicit(mockserver):
+    result = _run(
+        input="https://a.example", mockserver=mockserver, cli_params=["--intype", "txt"]
+    )
+    assert not result.returncode
+    assert (
+        result.stdout
+        == b'{"url": "https://a.example", "browserHtml": "<html><body>Hello<h1>World!</h1></body></html>"}\n'
+    )
+
+
+def test_intype_jsonl_implicit(mockserver):
+    result = _run(
+        input='{"url": "https://a.example", "browserHtml": true}', mockserver=mockserver
+    )
+    assert not result.returncode
+    assert (
+        result.stdout
+        == b'{"url": "https://a.example", "browserHtml": "<html><body>Hello<h1>World!</h1></body></html>"}\n'
+    )
+
+
+def test_intype_jsonl_explicit(mockserver):
+    result = _run(
+        input='{"url": "https://a.example", "browserHtml": true}',
+        mockserver=mockserver,
+        cli_params=["--intype", "jl"],
+    )
+    assert not result.returncode
+    assert (
+        result.stdout
+        == b'{"url": "https://a.example", "browserHtml": "<html><body>Hello<h1>World!</h1></body></html>"}\n'
+    )
+
+
+@pytest.mark.flaky(reruns=16)
```

Comment:
why is it flaky?

Reply:
Because there is a 50/50 chance of shuffle keeping the original order.
We could alternatively not use subprocess, and do some mocking to force `shuffle` to do `reversed`.
Thoughts?